modules/drivers/oracle/src/metabase/driver/oracle.clj

```diff
@@ -205,6 +212,27 @@
    (driver.common/start-of-week-offset driver)
    (partial hsql/call (u/qualified-name ::mod))))
 
+(defmethod sql.qp/->honeysql [:oracle :convert-timezone]
+  [driver [_ arg target-timezone source-timezone]]
+  (let [expr          (sql.qp/->honeysql driver arg)
+        has-timezone? (hx/is-of-type? expr #"timestamp(\(\d\))? with time zone")]
```

Comment:
Possible suggestion: if you replace this line, you could remove the changes to `hx/is-of-type?`, and replace it with `hx/is-of-database-type` in a separate PR.
```suggestion
        has-timezone? (re-find #"timestamp(\(\d\))? with time zone" (some-> expr hx/type-info hx/type-info->db-type str/lower-case))
```

Reply:
I don't see any downside to supporting regex for `is-of-type?`.
Maybe the real improvement would be moving from database-type to base-type. but that's the story for another day.